models/src/anemoi/models/samplers/diffusion_samplers.py

```diff
@@ -0,0 +1,255 @@
+# (C) Copyright 2024 Anemoi contributors.
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# In applying this licence, ECMWF does not waive the privileges and immunities
+# granted to it by virtue of its status as an intergovernmental organisation
+# nor does it submit to any jurisdiction.
+
+
+from typing import Callable
+from typing import Optional
+
+import torch
+from torch.distributed.distributed_c10d import ProcessGroup
+
+DenoisingFunction = Callable[
+    [torch.Tensor, torch.Tensor, torch.Tensor, Optional[ProcessGroup], Optional[list]], torch.Tensor
+]
+
+
+def get_noise_schedule(
```

Comment:
Have we tested this in anemoi-inference, do we need changes in the runner too to access the sampler or how will it work?